mm2src/coins/z_coin.rs

```diff
@@ -740,22 +751,37 @@ impl<'a> UtxoCoinBuilderCommonOps for ZCoinBuilder<'a> {
     fn ticker(&self) -> &str { self.ticker }
 }
 
-#[async_trait]
-impl<'a> UtxoFieldsWithIguanaPrivKeyBuilder for ZCoinBuilder<'a> {}
+impl<'a> UtxoFieldsWithSecp256k1SecretBuilder for ZCoinBuilder<'a> {}
+
+/// Although, `ZCoin` doesn't support [`PrivKeyBuildPolicy::Trezor`] yet,
+/// `UtxoCoinBuilder` trait requires `UtxoFieldsWithHardwareWalletBuilder` to be implemented.
+impl<'a> UtxoFieldsWithHardwareWalletBuilder for ZCoinBuilder<'a> {}
 
 #[async_trait]
-impl<'a> UtxoCoinWithIguanaPrivKeyBuilder for ZCoinBuilder<'a> {
+impl<'a> UtxoCoinBuilder for ZCoinBuilder<'a> {
     type ResultCoin = ZCoin;
     type Error = ZCoinBuildError;
 
-    fn priv_key(&self) -> &[u8] { self.secp_priv_key }
+    fn priv_key_policy(&self) -> PrivKeyBuildPolicy { self.priv_key_policy.clone() }
 
     async fn build(self) -> MmResult<Self::ResultCoin, Self::Error> {
-        let utxo = self.build_utxo_fields_with_iguana_priv_key(self.priv_key()).await?;
+        let utxo = self.build_utxo_fields().await?;
         let utxo_arc = UtxoArc::new(utxo);
 
-        let (_, my_z_addr) = self
-            .z_spending_key
+        let z_spending_key = match self.z_spending_key {
+            Some(ref z_spending_key) => z_spending_key.clone(),
+            None => match utxo_arc.priv_key_policy {
+                PrivKeyPolicy::KeyPair(key_pair) => ExtendedSpendingKey::master(key_pair.private_ref()),
```

Comment:
In GlobalHD case, `Zcoin` private key should be derived differently: https://zips.z.cash/zip-0032. You can use wallets from https://z.cash/wallets/ for testing. At least, I got the same address for a single seed from Ywallet and Zecwallet Lite. Our implementation should be compatible with them.

Reply:
As we discussed DM, I returned `PrivKeyBuildPolicy::GlobalHDAccount` enum variant to handle it correctly at the `ZCoin` coin builder.
Please also note that there are two different derivation paths for ZCoin: https://github.com/KomodoPlatform/atomicDEX-API/blob/mm2-hd-account/mm2src/mm2_test_helpers/src/for_tests.rs#L317 and https://github.com/KomodoPlatform/atomicDEX-API/blob/mm2-hd-account/mm2src/mm2_test_helpers/src/for_tests.rs#L313

This may be needed if we decide to support t-addresses at some point. Otherwise `m/32'/...` derivation path will be used to derive a secp256k1 key-pair (which is currently used at the swaps).